lib/common/client/base.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require 'faraday'
+require 'common/client/errors'
+
+require 'common/exceptions/external/backend_service_exception'
+
+module Common
+  module Client
+    ##
+    # Raised if the Faraday adapter is a Faraday::Adapter::HTTPClient and the
+    # Common::Client::Middleware::Request::RemoveCookies middleware in not included.
+    #
+    class SecurityError < StandardError
+    end
+
+    ##
+    # Raised when the breakers Faraday middleware is not first in the stack.
+    #
+    class BreakersImplementationError < StandardError
+    end
+
+    ##
+    # Base class for creating HTTP services. Wraps the Faraday gem and is configured via by passing in a
+    # {Common::Client::Configuration::REST} or {Common::Client::Configuration::SOAP} depending on the type
+    # of service you're connecting to. Once configured requests are made via the `perform` method.
+    #
+    # @example Create a service and make a GET request
+    #   class MyService < Common::Client::Base
+    #     configuration MyConfiguration
+    #
+    #     def get_resource
+    #       perform(:get, '/api/v1/resource')
+    #     end
+    #   end
+    #
+    #   service = MyService.new
+    #   response = service.get_resource
+    #
+    # @example a POST request with a body, headers, and Faraday options
+    #   def post_resource(json)
+    #     headers = { 'Content-Type' => 'application/json' }
+    #     options = { timeout: 60 }
+    #     response = perform(:post, '/submit', json, headers, options)
+    #   end
+    #
+    class Base
+      ##
+      # Sets the configuration singleton to use
+      #
+      def self.configuration(configuration = nil)
+        @configuration ||= configuration.instance
+      end
+
+      private
+
+      def config
+        self.class.configuration
+      end
+
+      def connection
+        @connection ||= lambda do
+          connection = config.connection
+          # handlers = connection.builder.handlers
+
+          # not using HTTPClient
+          # if handlers.include?(Faraday::Adapter::HTTPClient) &&
+          #   !handlers.include?(Common::Client::Middleware::Request::RemoveCookies)
+          #  raise SecurityError, 'http client needs cookies stripped'
+          # end
+
+          # not using breakers for Redis at this time
+          # if handlers.include?(Breakers::UptimeMiddleware)
+          #  return connection if handlers.first == Breakers::UptimeMiddleware
+          #
+          #  raise BreakersImplementationError, 'Breakers should be the first middleware implemented.'
+          # else
+          #  warn("Breakers is not implemented for service: #{config.service_name}")
+          # end
+
+          connection
```

Comment:
If you're not using code, delete it rather than comment it out will make for a cleaner history..

Reply:
removed